Event table update

guangguangzai opened the pull request:
Added 3 functions of generate event tables; added 2 tests.

guangguangzai committed:
generate_event_table_update

guangguangzai committed:
Style code (GHA)

guangguangzai committed:
update function name

guangguangzai committed:
update functions

guangguangzai committed:
Style code (GHA)

guangguangzai committed:
format adjustment

guangguangzai committed:
Merge branch 'Event_Table' of https://github.com/Merck/wpgsd into Event_Table

guangguangzai committed:
Style code (GHA)

guangguangzai committed:
changed intersection strings

guangguangzai committed:
change intersection symbol

LittleBeannie commented on R/generate_event_cc_addhypo.R:
```diff
@@ -0,0 +1,64 @@
+#' )
+#'
+#' result_table <- generate_event_table_cc(input_data, hypothesis)
+#' sorted_data <- result_table[order(result_table$analysis), ]
```
It looks like the output of `generate_event_table_cc` is sorted. Could you please check if there is a specific reason why the `sorted_data` is required in the example?

LittleBeannie commented on R/generate_event_ol_addhypo.R:
```diff
@@ -0,0 +1,67 @@
+#' generate event table in overall population
```
Please change the file name same as the function name.

LittleBeannie commented on R/generate_event_ol_addhypo.R:
```diff
@@ -0,0 +1,67 @@
+#' generate event table in overall population
+#'
+#' @param input_data input data in overall population situation
```
Please be more specific of `input_data` and `hypothesis`.

LittleBeannie commented on R/generate_event_table_1.R:
```diff
@@ -0,0 +1,74 @@
+#'
+#' @param input_data user input of population at IA and FA
+#' @param hypothesis Illustrate groups want to compare
+#' @param type either common control or overlap population
```
Please update to @param type Either `"common_control"` or `"overlap_population"`

LittleBeannie commented on R/generate_event_table_1.R:
```diff
@@ -0,0 +1,74 @@
+#' Generate table of event counts from user input, including two conditions: common control and overlap population
+#'
+#'
+#' @param input_data user input of population at IA and FA
```
Please be more specific of `input_data` and `hypothesis`.

LittleBeannie commented on R/generate_event_table_1.R:
```diff
@@ -0,0 +1,74 @@
+#' # Generate event table using the updated function
+#' result_table <- generate_event_table(input_data, hypothesis, type = "overlap_population")
+#'
+#' # Sort the result table by analysis
```
Echo with my previous comment: could you please check if the sort required?

LittleBeannie commented on R/generate_event_table_1.R:
```diff
@@ -0,0 +1,74 @@
+  } else if (type == "overlap_population") {
+    result_df <- generate_event_table_ol(input_data, hypothesis) # see generate_event_ol_addhypo.R
+  }
+  result_df <- result_df[!duplicated(result_df), ]
```
It looks like the output is already sorted in `generate_event_table_cc` and `generate_event_table_ol`. Is the line 71-73 may duplicated?

LittleBeannie reviewed (commented):
Hi @guangguangzai, thank you for moving this forward! A lot of work has been done. Please take a look at the following 19 comments before our meeting and get the east ones fixed. Then we could focus on the more challenging comments when we meet this week.

guangguangzai commented on R/generate_event_cc_addhypo.R:
```diff
@@ -0,0 +1,64 @@
+#' generate event table in common control
```
changed to generate_event_table_cc

guangguangzai commented on R/generate_event_cc_addhypo.R:
```diff
@@ -0,0 +1,64 @@
+#' generate event table in common control
+#'
+#' @param input_data input data in common control situation
```
updated

guangguangzai commented on R/generate_event_cc_addhypo.R:
```diff
@@ -0,0 +1,64 @@
+#' @param input_data input data in common control situation
+#' @param hypothesis comparison group
+#'
+#' @return
```
added

guangguangzai committed:
update skeleton

guangguangzai committed:
update result_df

guangguangzai committed:
update base on comments collected

guangguangzai committed:
Style code (GHA)

LittleBeannie commented on R/generate_event_cc_addhypo.R:
```diff
@@ -1,15 +1,23 @@
-#' generate event table in common control
+#' generate_event_table_cc
```
Please delete this line.

LittleBeannie commented on R/generate_event_cc_addhypo.R:
```diff
@@ -1,15 +1,23 @@
 #'
-#' @return
-#' @export
+#' @param input_data A dataframe containing at least two columns: one for the population (experimental arms and control) 
```
Please specify the column name of this dataframe.

LittleBeannie commented on R/generate_event_cc_addhypo.R:
```diff
@@ -1,15 +1,23 @@
+#' @param input_data A dataframe containing at least two columns: one for the population (experimental arms and control) 
+#'                   and one or more columns for analyses (e.g., interim and final analyses).The analysis columns must be numerical values 
+#'                   representing the number of events observed during the interim and/or final analyses.
+#' @param hypothesis A list containing hypotheses specifying comparisons between experimental arms and the control group, 
```
"A list containing hypotheses" should be good.

LittleBeannie commented on R/generate_event_ol_addhypo.R:
```diff
@@ -1,15 +1,33 @@
-#' generate event table in overall population
+#' generate_event_table_ol
```
Please delete this line

LittleBeannie commented on R/generate_event_table_1.R:
```diff
@@ -1,58 +1,61 @@
-#' Generate table of event counts from user input, including two conditions: common control and overlap population
+#' generate_event_table_1
```
Please delete this line.

guangguangzai committed:
delete old functions

guangguangzai committed:
delete old functions

guangguangzai committed:
Style code (GHA)

guangguangzai committed:
update eventdata to event

guangguangzai committed:
eventdata to event

Merge branch 'Event_Table' of https://github.com/Merck/wpgsd into Event_Table

# Conflicts:
#	R/generate_event_cc.R
#	R/generate_event_ol.R
#	R/generate_event_table_.R
#	tests/testthat/test-developer-generate_event_commoncontrol_addhypo.R
#	tests/testthat/test-developer-generate_event_population_addhypo.R

guangguangzai committed:
eventdata to event

guangguangzai committed:
Style code (GHA)

guangguangzai committed:
update functions and tests

guangguangzai committed:
Style code (GHA)

LittleBeannie reviewed (approved):
Thanks, @guangguangzai! Nice job!
